### PR TITLE
DataViews: set proper semantics for dropdown items

### PR DESCRIPTION
**Pull request by oandregal**
Part of https://github.com/WordPress/gutenberg/issues/55083

## What?

Set the proper semantics for the existing menu items.

## Why?

In some cases, the item was declared as behaving as a checkbox while their behavior was actually like a radio item.

## How?

Set the proper `role` for every `DropdownMenuItem`.

### CODE REVIEW

**Comment by ntsekouras** [comment]
Thanks for the PR! I know @ciampo  has a PR for replacing the dropdown, but maybe it's better to use the new one and make improvement there if needed?

**Comment by oandregal** [comment]
> Thanks for the PR! I know @ciampo has a PR for replacing the dropdown, but maybe it's better to use the new one and make improvement there if needed?

No reason not to land this simple fix while the new dropdown is being worked on (I don't know the timeline for that, we may find issues while doing it, etc.).

**Review by ntsekouras — APPROVED** [review]

**Review by ciampo — COMMENTED** [review]
Thank you for working on this!

Rather than using `DropdownMenuItem` and manually adding `role` and `aria-checked` attributes, we should use the dedicated menu radio item components.

- If we're using the `DropdownMenuV2` (the radix-based one, that will soon be deleted), we should use the `value` / `onValueChanged` props instead of the `onSelect` prop ([example](https://github.com/WordPress/gutenberg/blob/4b10f653ba12baebd374dacdb2df2d93c017a422/packages/components/src/dropdown-menu-v2/stories/index.story.tsx#L124-L141))
- If we were to use the new version of the component (based on `ariakit`), we should use the `checked` and `onChange` props on each menu items (you can see how I approach a similar refactor in #55625)

Hope this makes sense!

**Comment by oandregal** [comment]
@ciampo Thanks for the advice, that's great.

There are some subtle differences between the new components and the old ones (focus, for example), so I'd rather update them all together. I've tried using the specific components in other PRs (radio, checkbox) and ran into issues (couldn't uncheck, etc.) — perhaps it was my own fault, I'm not sure, I'll revisit.

Anyway, this PR also makes the e2e test more robust (searching string using exact, delete template in `afterAll` for it not to be affected by test failures, etc.). If it's not too much trouble, I'm going to go ahead with this and then will look into updating your PR with the latest changes (rebase due to code reorganization) https://github.com/WordPress/gutenberg/pull/55625

**Comment by ciampo** [comment]
> If it's not too much trouble, I'm going to go ahead with this and then will look into updating your PR with the latest changes

Sounds good

> tried using the specific components in other PRs (radio, checkbox) and ran into issues (couldn't uncheck, etc.)

That is possible, and it would be because radio items are not supposed to be unchecked (see [related comment](https://github.com/WordPress/gutenberg/pull/55625#discussion_r1376355788)). For posterity, I'm also linking to a [related conversation](https://github.com/WordPress/gutenberg/pull/55625#discussion_r1376355788) that we've been having about the right semantics / widgets that should be probably used for these components.

File: packages/edit-site/src/components/dataviews/add-filter.js [hunk 1]
```diff
@@ -81,6 +81,8 @@ export default function AddFilter( { fields, view, onChangeView } ) {
 						{ filter.elements.map( ( element ) => (
 							<DropdownMenuItem
 								key={ element.value }
+								role="menuitemradio"
+								aria-checked={ false }
```
Comment (andrewhayward) [inline packages/edit-site/src/components/dataviews/add-filter.js:85]: This is a weird one, but I'd probably argue that these should have a role of `menuitem`. They are not values in and of themselves; instead, they are all actions that create a new filter with a particular value.

```suggestion
								role="menuitem"
```

Ideally we'd put this in front of actual users, but given the submenu isn't available once the filter has been created, it kind of ceases to matter what their selected state is, or whether they are mutually exclusive.

Comment (andrewhayward) [reply packages/edit-site/src/components/dataviews/add-filter.js:85]: (Looks like I was 30 seconds late with feedback 😩 Not a blocker, just something to think about.)

Comment (oandregal) [reply packages/edit-site/src/components/dataviews/add-filter.js:85]: Ugh, sorry, had this to "auto-merge" once test pass. Feedback is never late, though :) https://github.com/WordPress/gutenberg/pull/56714

Also incorporated this into the refactor to ariakit: https://github.com/WordPress/gutenberg/pull/55625

**Pull request merged**